src/peft/tuners/adaption_prompt/utils.py

```diff
@@ -95,7 +95,12 @@ def llama_compute_query_states(model: nn.Module, **kwargs) -> torch.Tensor:
             new_cache_positions = torch.arange(past_seen_tokens, past_seen_tokens + q_len, device=value_states.device)
         position_ids = new_cache_positions.unsqueeze(0)
 
-    cos, sin = model.rotary_emb(value_states, seq_len=q_len + past_seen_tokens, position_ids=position_ids)
+    rotary_emb_kwargs = {"position_ids": position_ids}
+    # The `seq_len` argument has been officially deprecated in trasformers >= 4.39.0
```

Comment:
```suggestion
    # The `seq_len` argument has been officially removed in transformers >= 4.39.0
```

It's been deprecated before, but removed now.

Reply:
ah yes correct!